developers/advanced/ingestion-level-aggregations.md

```diff
@@ -0,0 +1,151 @@
+# Ingestion Aggregations
+
+Many data analytics use-cases only need aggregated data. For example, data used in charts can be aggregated down to one row per time bucket per dimension combination. 
+
+Doing this results in much less storage and better query performance. Configuring this for a table is done via the Aggregation Config in the [table config](../../configuration-reference/table.md).
+
+## Aggregation Config
+
+The aggregation config controls the aggregations that happen during realtime data ingestion. Offline aggregations must be handled separately. 
+
+Below is a description of the config, which is defined in the ingestion config of the table config. 
+
+```json
+{
+  "tableConfig": {
+    "tableName": "...",
+    "ingestionConfig": {
+      "aggregationConfigs": [{
+        "columeName": "aggregatedFieldName",
+        "aggregationFunction": "<aggregationFunction>(<originalFieldName>)"
+      }]
+    }
+  }
+}
+```
+
+## Example Scenario
+Here is an example of sales data, where only the daily sales aggregates per product are needed. 
+
+### Schema
+
+```json
+{
+  "schemaName": "daily_sales_schema",
+  "dimensionFieldSpecs": [
+    {
+      "name": "product_name",
+      "dataType": "STRING"
+    }
+  ],
+  "metricSpecs": [
+    {
+      "name": "sales_count",
+      "dataType": "LONG"
+    },
+    {
+      "name": "total_sales",
+      "dataType": "DOUBLE"
+    }
+  ],
+  "dateTimeFieldSpecs": [
+    {
+      "name": "daysSinceEpoch",
+      "dataType": "LONG",
+      "format": "1:MILLISECONDS:EPOCH",
+      "granularity": "1:MILLISECONDS"
+    }
+  ]
+}
+```
+
+### Table Config
+
+```json
+{
+  "tableName": "daily_sales",
+  "ingestionConfig": {
+    "transformConfigs": [
+      {
+        "columnName": "daysSinceEpoch",
+        "transformFunction": "toEpochDays(timestamp)"
+      }
+    ],
+    "aggregationConfigs": [
+      {
+        "columnName": "total_sales",
+        "aggregationFunction": "SUM(price)"
+      },
+      {
+        "columnName": "sales_count", 
+        "aggregationFunction": "COUNT(*)"
+      }
+    ]
+  }
+}
+```
+
+### Example Input Data
```

Comment:
nit: I would put the example data above the schema and explicitly call out that the schema should have the final fields